serverlessworkflow/modules/ROOT/pages/use-cases/advanced-developer-use-cases/deployments/deploying-sonataflow-management-console-on-openshift.adoc

```diff
@@ -0,0 +1,149 @@
+
+
+This document describes how to deploy the {product_name} Management Console using on OpenShift.
+
+The SonataFlow Management Console is a web interface designed to manage and monitor workflows in a production-ready environment. It allows users to initiate workflows, trigger Cloud Events, and monitor their execution.
+
+.Prerequisites
+* Your xref:use-cases/advanced-developer-use-cases/deployments/deploying-on-openshift.adoc[{product_name} application] is deployed and ready to use.
+* link:{ocp_cli_url}[OpenShift CLI] is installed.
+* (Optional) xref:security/orchestrating-third-party-services-with-oauth2.adoc[Keycloak server] is installed in OpenShift.
+
+== Set up {ocp_name} and deploy your SonataFlow application on OpenShift
+
+The SonataFlow Management Console relies on the underlying services of the SonataFlow application, including the Data Index service. The Data Index provides data from workflow executions, enabling the Management Console to display workflow information. 
+
+Ensure that your Data Index service is deployed and accessible before deploying the Management Console.
+Instructions to set up {ocp_name} and deploy your SonataFlow application on OpenShift can be found xref:use-cases/advanced-developer-use-cases/deployments/deploying-on-openshift.adoc[here].
+
+== (Optional) Deploy Keycloak for production
+
+_If you already have Keycloak deployment or server available you can skip this section._
+
+For production environments, deploy Keycloak to handle authentication or choose your orwn OAuth2 server. You can refer to the SonataFlow Keycloak documentation xref:security/orchestrating-third-party-services-with-oauth2.adoc[here].
+
+== Deploy {management_console}
+
+Create the deployment for the SonataFlow Management Console by applying the following YAML definition:
+
+.Example configuration in `osl-management-console-deployment.yaml`
+[source,yaml,subs="attributes+"]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: osl-management-console
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: osl-management-console
+  template:
+    metadata:
+      labels:
+        app: osl-management-console
+    spec:
+      containers:
+        - name: osl-management-console
+          image: {sonataflow_management_console_imagename}:latest
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+              name: http
+              protocol: TCP
+          env:
+            - name: SONATAFLOW_MANAGEMENT_CONSOLE_KOGITO_ENV_MODE
+              value: DEV <1>
+            - name: SONATAFLOW_MANAGEMENT_CONSOLE_DATA_INDEX_ENDPOINT
+              value: DATA-INDEX-URL <2>
+----
+<1> (Optional) Environment mode: "PROD" or "DEV". PROD enables Keycloak integration.
+<2> The URL to the Data Index Deployment created xref:use-cases/advanced-developer-use-cases/deployments/deploying-on-openshift.adoc[here].
```

Comment:
@fantonangeli I had to expose the DI URL too, should we mention that?

Reply:
Yes, we must mention it.